blockchain/types/account/externally_owned_account.go

```diff
@@ -67,3 +116,194 @@ func (e *ExternallyOwnedAccount) Equal(a Account) bool {
 
 	return e.AccountCommon.Equal(e2.AccountCommon)
 }
+
+func (e *ExternallyOwnedAccount) GetStorageRoot() common.ExtHash {
+	return e.storageRoot
+}
+
+func (e *ExternallyOwnedAccount) GetCodeHash() []byte {
+	return e.codeHash
+}
+
+func (e *ExternallyOwnedAccount) GetCodeFormat() params.CodeFormat {
+	return e.codeInfo.GetCodeFormat()
+}
+
+func (e *ExternallyOwnedAccount) GetVmVersion() params.VmVersion {
+	return e.codeInfo.GetVmVersion()
+}
+
+func (e *ExternallyOwnedAccount) SetStorageRoot(h common.ExtHash) {
+	e.storageRoot = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeHash(h []byte) {
+	e.codeHash = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeInfo(ci params.CodeInfo) {
+	e.codeInfo = ci
+}
+
+func (e *ExternallyOwnedAccount) toSerializable() *externallyOwnedAccountSerializable {
+	return &externallyOwnedAccountSerializable{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot.Unextend(),
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) toSerializableExt() *externallyOwnedAccountSerializableExt {
+	return &externallyOwnedAccountSerializableExt{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot,
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) fromSerializable(o *externallyOwnedAccountSerializable) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot.ExtendZero()
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) fromSerializableExt(o *externallyOwnedAccountSerializableExt) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLP(w io.Writer) error {
+	if e.isEOAWithCode() {
+		return rlp.Encode(w, e.toSerializable())
+	}
+	return rlp.Encode(w, e.AccountCommon.toSerializable())
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLPExt(w io.Writer) error {
```

Comment:
How about using switch statement?

For reference, but I haven't actually tested whether it works...
```
func (e *ExternallyOwnedAccount) EncodeRLPExt(w io.Writer) error {
	var serializable interface{}

	switch {
	case e.isEOAWithCode() && e.storageRoot.IsZeroExtended():
		serializable = e.toSerializable() // EOA without code. [n, b, hR, k, 0x0, eCH, 0x0]
	case e.isEOAWithCode():
		serializable = e.toSerializableExt() // EOA with code, has ExtHash [n, b, hR, k, sRExt, cH, cI]
	default:
		serializable = e.AccountCommon.toSerializable() // EOA without code. [n, b, hR, k]
	}

	return rlp.Encode(w, serializable)
}
```